lib/elixir/test/elixir/code_test.exs

```diff
@@ -55,6 +55,21 @@ defmodule CodeTest do
                end)
     end
 
+    test "reports correct position for duplicate struct fields" do
```

Comment:
Instead of adding a warning here, can you please make sure that we assert on the correct file/line inside warnings_test? We may need to make it something like `nofile:2: TestMod`.

Reply:
That's indeed better, thanks for the suggestion!